### DISCUSSION TITLE
Render from matching active object

### DISCUSSION THREAD

**Discussion by EdmisP** [Gorgious56/asset_browser_utilities#119] [Q&A]
Is there a way to render asset preview from partially matching its name object.
For example i have material node groups as assets and want to generate previews from objects with partially matching names  as active object.

**Comment by Gorgious56** [comment]
Hello ! Could you add a more precise example ? I'm not sure I understand 100% what you want to achieve ?

**Comment by EdmisP** [comment]
![RenderActiveObject](https://user-images.githubusercontent.com/10084761/221173804-c448dcd9-aa2b-4f0c-a872-18df5ac056d7.jpg)
I have meshes with diferente color pallet color materials assigned (Green). Materials have that color as group node and marked as asset, so it could be used on multiple  different type materials if needed (Red). But Node groups have no auto preview. 
So maybe there is a way to cycle through selected Assets in asset browser and select as active objects meshes with similar names (in this example its 13 first characters) and and do "Render active object".

**Comment by Gorgious56** [comment]
Alright, I understand. Though it's a bit niche so I can't promise I'll work on implement it in the near future. I'll keep the thread open for when I have more time to work on this addon. I'll keep you posted.

Cheers

**Reply by EdmisP** [reply]
Thank you for consideration. Until then i will try to get some form of this script from ChatGPT. All previous attempts failed 😄
